### DISCUSSION TITLE
How to easily queue up an entire playlist/album? And can this be done with and without shuffling?

### DISCUSSION THREAD

**Discussion by sheriferson** [rekkyrosso/ampcast#17] [General]
Let's say I'm looking at a Plex playlist, or an Apple Music playlist. I'd like to play the entire playlist, sometimes in order and sometimes shuffled depending on the playlist.

What would be nice is to double click on the playlist, or press a shortcut, and have all its tracks immediately added to the queue.

Right now as far as I know the only way to do this is select the playlist, move focus to the list of tracks within the playlist and select all, then <kbd>Enter</kbd> to queue or <kbd>⌘-Enter</kbd>to play now. It would be nice for that to be less manual to do.

Maybe the <kbd>Enter</kbd> and <kbd>⌘-Enter</kbd> shortcuts can be extended to items in the Playlist view such that invoking that with the playlist name selected does the same thing as selecting-all on tracks and then pressing the same thing?

**Comment by rekkyrosso** [comment]
You can already add a whole album by dragging it, or pressing  `Enter` or double-clicking.

Playlists are a bit trickier because they might be really big (thousands of tracks). That might mean dozens of requests.

I do intend to add better support for adding an entire playlist. But I need to think through the UX before I change anything.

**Comment by rekkyrosso** [comment]
You can now add a playlist the same as you can with albums. 

From the release notes:

> Playlists can be queued (the same as albums), except for very large playlists. If you can see a progress bar then the playlist is not yet fully playable.

As for shuffling, I'll maybe add a "Shuffle selection" option to the playlist menu. There's already a "Reverse selection" option.
